src/Dynamo/src/Dynamo.CSLang/CSNamespace.cs

```diff
@@ -0,0 +1,27 @@
+// Copyright (c) Microsoft Corporation.
```

Comment:
Isn't this part of the emitter's environment? This should be known by parsing the ABI.

Reply:
It will be known by the parsing ABI, but Dynamo is a structured output generator and knows nothing about the parser. What will happen is at the beginning of working on a type, the environment for a dynamo emitter will have a `CSFile` member that the parsing ABI will populate with the namespace. As for the usings, they are still needed and can either be done just after CSFile creation or during the process of binding. Every time a C# type is referenced, a simple AddIfNotPresent gets called, ensuring that (1) you never miss a using and (2) you never duplicate a using. This also holds true for types that are not directly referenced by the ABI, for example if we need `System.Runtime` or `System.Runtime.InteropServices` etc.

Reply:
Currently, the Dynamo contains emitter and context for marshalling. I would like to see it decoupled, and I am interested to see how it evolves in the future.